openeo/extra/job_management/__init__.py

```diff
@@ -535,13 +560,18 @@ def _job_update_loop(
                     to_add = self.backends[backend_name].parallel_jobs - backend_load
                     for i in not_started.index[total_added : total_added + to_add]:
                         self._launch_job(start_job, df=not_started, i=i, backend_name=backend_name, stats=stats)
-                        stats["job launch"] += 1
-
                         job_db.persist(not_started.loc[i : i + 1])
+
+                        stats["job launch"] += 1
```

Comment:
I would not move stat updates like this. Keep them as close as possible to the actual event they are about.